src/index.js

```diff
@@ -1,6 +1,32 @@
-/**
-* Put your JavaScript here
-*/
+let state = {
+  todos: [
+    { title: 'First todo', complete: false },
+    { title: 'Second todo', complete: true },
+    { title: 'Third todo', complete: true },
+  ],
+};
+
+document.onload = init();
+
+function init() {
+	const listNode = document.getElementById('list');
+
+	const filterButton = document.getElementById('filter-button')
+	filterButton.addEventListener('click', (event) => {
+		const items = document.querySelectorAll('#list li.todo--complete');
+		items.forEach( item => item.classList.toggle('hide') );
+	})
+
+	state.todos.forEach( item => addTodoNode(item) )
+
+	function addTodoNode ({title, complete}) {
+		let item = document.createElement('li');
+		item.textContent = title;
+		if (complete === true) item.classList.add('todo--complete');
+		item.addEventListener('click', event => event.target.classList.toggle('todo--complete') );
```

Comment:
It looks like we may be getting out of sync between what's in the DOM and what's in the state, right? If I mark one as complete, it will _look_ complete, but in our state it will still be incomplete.

You could just toggle the flag in the callback there, but then you have to manually keep the two in sync. What might be a better way?